src/App/Page.php

```diff
@@ -78,10 +81,11 @@ class Page implements ArrayAccess
 	/**
 	 * @param string $basepath The Page basepath
 	 */
-	public function __construct(string $basepath)
+	public function __construct(string $basepath, EventDispatcherInterface $eventDispatcher)
 	{
 		$this->timestamp = microtime(true);
 		$this->basePath = $basepath;
+		$this->eventDispatcher = $eventDispatcher;
```

Comment:
I assume that at this point we expect the EventDispatcher instance to be the one that was created in `App->registerEventDispatcher` thanks to Dice magic?

Reply:
Correct, we can simply add any dependency in constructor and Dice will inject them using autowiring.